src/rules.c

```diff
@@ -389,13 +389,15 @@ int SchemaRule_RdbLoad(IndexSpec *sp, RedisModuleIO *rdb, int encver) {
   RSLanguage lang_default = LoadUnsigned_IOError(rdb, goto cleanup);
 
   QueryError status = {0};
-  SchemaRule *rule = SchemaRule_Create(&args, sp, &status);
+  SchemaRule *rule = SchemaRule_Create(&args, ref, &status);
   if (!rule) {
     RedisModule_LogIOError(rdb, "warning", "%s", QueryError_GetError(&status));
     RedisModule_Assert(rule);
   } else {
     rule->score_default = score_default;
     rule->lang_default = lang_default;
+    // No need to validate the reference here, since we are loading it from the RDB
+    IndexSpec *sp = StrongRef_Get(ref);
     sp->rule = rule;
```

Comment:
Ohh right, I see the spec holds the rule and the rule holds the spec. this is a classic weak ref usecase.

Reply:
check out #3352, it might be a better fix